sklearn/decomposition/tests/test_fastica.py

```diff
@@ -167,7 +167,7 @@ def g_test(x):
         with pytest.raises(ValueError):
             ica.fit(m.T)
 
-    with pytest.raises(TypeError):
```

Comment:
We can remove this part and on the check just above, we don't need to check for the string. It is covered by the common test.

Then, we want to test for passing a wrong callable apparently. In this case, we need to check specifically the error message with the keyword `match` to be sure that we check what we expect as error.